src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -151,5 +156,47 @@ static CodeAction GetOptionalContructorParametersCodeAction(Document document, C
                     document, constructorCandidate, containingType, missingOptionalParameters, useSubMenuName);
             }
         }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(Document priorDocument, TextSpan priorSelection, Document currentDocument, string serializedIntentData, CancellationToken cancellationToken)
+        {
+            var addConstructorParametersResult = await AddConstructorParametersFromMembersAsync(priorDocument, priorSelection, cancellationToken).ConfigureAwait(false);
+            if (addConstructorParametersResult == null)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            var actions = addConstructorParametersResult.Value.RequiredParameterActions.Concat(addConstructorParametersResult.Value.OptionalParameterActions);
+            if (actions.IsEmpty)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            using var _ = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+            foreach (var action in actions)
+            {
+                var changedSolution = await action.GetChangedSolutionInternalAsync(postProcessChanges: true, cancellationToken).ConfigureAwait(false);
+                var intent = new IntentProcessorResult(changedSolution, action.Title, action.ActionName);
+                results.Add(intent);
+            }
+
+            return results.ToImmutable();
+        }
+
+        private struct AddConstructorParameterResult
```

Comment:
```suggestion
        private readonly struct AddConstructorParameterResult
```

also, consider breaking into its own file.